src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -165,32 +161,12 @@ public void LogEvent(string eventName, string functionName = null, string data =
 
         internal void FunctionStarted(FunctionStartedEvent startedEvent)
         {
-            lock (_functionActivityTrackerLockObject)
```

Comment:
I have no idea why the code was written in the horribly inefficient way it was previously. On each request, we were locking and creating a new static tracker instance.